crates/node/src/runtime/dag_module.rs

```
@@ -350,30 +345,45 @@ impl Handler<EventMessage> for DagModule {
-                                    block.get_header().round,
-                                    block.get_header().epoch,
-                                    self.claim.clone(),
-                                ),
-                            ));
-                        }
-                        // Emit event for state update
+                let mut mine_block: Option<ConvergenceBlock> = None;
+                let block_hash = certificate.block_hash.clone();
+                if let Ok(Some(Block::Convergence { mut block })) =
+                    self.dag.write().map(|mut bull_dag| {
+                        bull_dag
+                            .get_vertex_mut(block_hash)
+                            .map(|vertex| vertex.get_data())
+                    })
+                {
+                    block.append_certificate(certificate.clone());
+                    self.last_confirmed_block_header = Some(block.get_header());
+                    mine_block = Some(block.clone());
+                }
+                if let Some(block) = mine_block {
+                    let proposal_block = Event::MineProposalBlock(
+                        block.hash.clone(),
+                        block.get_header().round,
+                        block.get_header().epoch,
+                        self.claim.clone(),
+                    );
+                    if let Err(err) = self
+                        .events_tx
+                        .send(EventMessage::new(None, proposal_block.clone()))
+                        .await
+                    {
+                        let err_msg = format!(
+                            "Error occurred while broadcasting event {proposal_block:?}\ndetails: {err:?}"
+                        );
+                        return Err(TheaterError::Other(err_msg));
                     }
+                } else {
+                    println!("Missing ConvergenceBlock for certificate: {certificate:?}");
```

@nopestack is displaying this helpful at all? Before, if this was `None` it went silently ignored.